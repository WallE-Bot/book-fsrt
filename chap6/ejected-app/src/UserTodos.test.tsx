import React from 'react';
import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import UserTodos from './UserTodos';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

describe("Test UserTodos", () => {
  // mock data
  /*
  const getUserFullname = async (username: string): Promise<string> => {
    const usersResponse = await fetch(
      `https://jsonplaceholder.typicode.com/users`
    );

    if(usersResponse.ok) {
      const users = await usersResponse.json();
      const userByName = users.find((usr: any) => {
        return usr.username.toLowerCase() === username;
      });
      return userByName.name;
    }
    return '';
  }
  */

  it('renders without crashing', () => {
    const username = 'testUser';

    const { baseElement } = render(<UserTodos
      username={username}
    />);

    expect(baseElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const username = 'testUser';

    const { baseElement } = render(<UserTodos
      username={username}
    />);

    expect(baseElement).toMatchSnapshot();
  });
})
