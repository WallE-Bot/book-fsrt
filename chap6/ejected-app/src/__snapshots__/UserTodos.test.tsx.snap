// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserTodos matches snapshot 1`] = `
<body>
  <div>
    <ul />
  </div>
</body>
`;
